### PR TITLE
OSX -> macOS

### PR DESCRIPTION
**Pull request by rex4539**
Replaced OSX with macOS in places I thought were appropriate and added a couple of small unification changes.

### CODE REVIEW

**Comment by Empact** [comment]
Related prior work: #13366

File: src/tinyformat.h [hunk 1]
```diff
@@ -149,13 +149,13 @@ namespace tfm = tinyformat;
...
 #if defined(__GLIBCXX__) && __GLIBCXX__ < 20080201
-//  std::showpos is broken on old libstdc++ as provided with OSX.  See
+//  std::showpos is broken on old libstdc++ as provided with macOS.  See
 //  http://gcc.gnu.org/ml/libstdc++/2007-11/msg00075.html
 #   define TINYFORMAT_OLD_LIBSTDCPLUSPLUS_WORKAROUND
 #endif
 
 #ifdef __APPLE__
-// Workaround OSX linker warning: Xcode uses different default symbol
+// Workaround macOS linker warning: Xcode uses different default symbol
```
Comment (MarcoFalke) [inline src/tinyformat.h]: Please report this upstream instead.

File: src/qt/guiutil.h [hunk 2]
```diff
@@ -234,7 +234,7 @@ namespace GUIUtil
     };
 
 #if defined(Q_OS_MAC)
-    // workaround for Qt OSX Bug:
+    // workaround for Qt macOS Bug:
```
Comment (MarcoFalke) [inline src/qt/guiutil.h]: Can be fixed up as part of #13362 or not at all.

**Comment by MarcoFalke** [comment]
Tend to NACK here for the comments given above. Imo, it would be more helpful if `contrib/gitian-build.sh` was rewritten to python to make it more user friendly. I don't see any increase in user-friendliness in this pull request as is.

File: src/tinyformat.h (hunk 1 above)
Comment (rex4539) [reply src/tinyformat.h]: https://github.com/c42f/tinyformat/pull/52

**Comment by rex4539** [comment]
Addressed the first comment. There is nothing I can do about the second comment :)

File: src/qt/guiutil.h (hunk 2 above)
Comment (fanquake) [reply src/qt/guiutil.h]: Should be going away in #13617. I can cherry-pick some of the other changes here into that.

**Comment by laanwj** [comment]
Agree with @marcofalke here on ~NACK. I think #13366 did the important changes here. We can forever keep creating PRs to change straggler occurrences in obscure places, but the added value is little compared to the overhead.

Also agree with converting `contrib/gitian-build.sh` to python.

**Pull request closed without merging**